exercises/Mathieu_Tillet/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * Skeleton main routine
+ */
+
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <vector>
+#include <algorithm>
+#include <tuple>
+
+using namespace std;
+
+int main(int argc, char **argv)
+{
+    map<string, int> word_count;
+
+    if (argc < 2){
+        cerr << "No file provided";
+        return -1;
+    }
+
+    // the input file
+    char *file_name = argv[1];
+
+    string out_name;
+    // Name of the file in which the output will be written
+    if (argc > 2){
+        // catch the second argument if one is provided
+        out_name = string(argv[2]);
+    } else {
+        // else create a new file
+        out_name = "./out.txt";
```

Comment:
In the wild, one should avoid hardcoded paths, especially containing `/` since it breaks the cross-platform compatibility; Windows uses `\`. In reality we would use a library to abstract out that aspect, such as `Poco::File` or `std::filesystem`.